[nnc][aot_compiler] Memory formats args to aot_compiler

IvanKobzarev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #72873

Differential Revision: [D34250984](https://our.internmc.facebook.com/intern/diff/D34250984)
```
build/bin/aot_model_compiler \
    --model $HOME/nnc/qsegm1002.ptl \
    --model_name=qsegm \
    --model_version=v1 \
    --input_dims="1,4,224,224" \
    --input_types="float" \
    --input_memory_formats="channels_last"
```

```
[DUMP kernel.cpp:1432] TensorExprKernel graph:
[DUMP kernel.cpp:1432] graph(%input.1 : Float(1, 4, 224, 224, strides=[200704, 1, 896, 4], requires_grad=0, device=cpu)):
[DUMP kernel.cpp:1432]   %1 : float[] = prim::Constant[value=[2., 2.]]()
...
```

IvanKobzarev committed:
[nnc][aot_compiler] Memory formats args to aot_compiler

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34250984).

priyaramani commented on binaries/aot_model_compiler.cc:
```diff
@@ -93,10 +101,14 @@ int main(int argc, char** argv) {
       "Number of input_dims and input_types should be the same");
+  const auto mem_formats_size = split(';', FLAGS_input_memory_formats).size();
+  CAFFE_ENFORCE(
+      mem_formats_size == 0 || mem_formats_size == dims_size,
```
Since we give a default value for FLAGS_input_memory_formats at line 35, `mem_formats_size == 0` will never be true. Did you mean to remove the default string in line 35 as we set default value in line 261, and this condition would also work?

IvanKobzarev committed:
Update on "[nnc][aot_compiler] Memory formats args to aot_compiler"


Differential Revision: [D34250984](https://our.internmc.facebook.com/intern/diff/D34250984)
```
build/bin/aot_model_compiler \
    --model $HOME/nnc/qsegm1002.ptl \
    --model_name=qsegm \
    --model_version=v1 \
    --input_dims="1,4,224,224" \
    --input_types="float" \
    --input_memory_formats="channels_last"
```

```
[DUMP kernel.cpp:1432] TensorExprKernel graph:
[DUMP kernel.cpp:1432] graph(%input.1 : Float(1, 4, 224, 224, strides=[200704, 1, 896, 4], requires_grad=0, device=cpu)):
[DUMP kernel.cpp:1432]   %1 : float[] = prim::Constant[value=[2., 2.]]()
...
```

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34250984).
